File: 2php/Memory.php
<?php
    class Memory extends Hardware{
        //atribut
        private $frequency = 0;
        private $memorySize = 0;
        private $supportsCuda = "blank";

        function __construct(){
        }
        
        //setter & getter
        function setfrequency($frequency){
            $this->frequency = $frequency;
        }
        function getfrequency(){
            return $this->frequency;
        }

        function setmemorySize($memorySize){
            $this->memorySize = $memorySize;
        }
        function getmemorySize(){
            return $this->memorySize;
        }

        function setsupportsCuda($supportsCuda){
            $this->supportsCuda = $supportsCuda;
        }
        function getsupportsCuda(){
            return $this->supportsCuda;
        }

        function __destruct(){
        }
    } 
?>

File: 2Python/Memory.py
from Hardware import Hardware

class Memory(Hardware):
    #private members
    __frequency = 0
    __memorySize = 0
    __supportsCuda = False

    def __init__(self):
        self.__frequency = 0
        self.__memorySize = 0
        self.__supportsCuda = False

    #getter & setter
    def setfrequency(self, frequency):
        self.__frequency = frequency

    def getfrequency(self):
        return self.__frequency
    
    def setmemorySize(self, memorySize):
        self.__memorySize = memorySize

    def getmemorySize(self):
        return self.__memorySize

    def setsupportsCuda(self, supportsCuda):
        self.__supportsCuda = supportsCuda

    def getsupportsCuda(self):
        return self.__supportsCuda
